subsys/bluetooth/host/hci_core.c

```diff
@@ -7778,6 +7779,11 @@ int bt_le_adv_start_legacy(const struct bt_le_adv_param *param,
 			  param->options & BT_LE_ADV_OPT_USE_IDENTITY);
 
 	return 0;
+
+failed:
+	adv_delete_legacy();
```

Comment:
Thank you for the fix, although I would suggest to do it a bit different.
In `bt_le_adv_start`:
```
	err = bt_le_adv_start_legacy(param, ad, ad_len, sd, sd_len);
	if (err) {
		adv_delete_legacy();
	}

	return err;
```

That is similar to how it is handled for extended advertise case.

Reply:
Sorry for reply late, I moved the adv instance allocate out of the le_adv_start_legacy(), so as to reuse and unify the adv_new_legacy() and adv_delete_legacy()